openedx/core/djangoapps/credit/partition_schemes.py

```diff
@@ -0,0 +1,98 @@
+"""
+Provides partition support to the user service.
+"""
+import logging
+from course_modes.models import CourseMode
+from student.models import CourseEnrollment
+from verify_student.models import SkippedReverification, VerificationStatus
+
+log = logging.getLogger(__name__)
+
+
+class VerificationPartitionScheme(object):
+    """
+    This scheme randomly assigns users into the partition's groups.
+    """
+    NON_VERIFIED_GROUP = 'non_verified_group'
+    VERIFIED_ALLOW_GROUP = 'verified_allow_group'
+    VERIFIED_DENY_GROUP = 'verified_deny_group'
+
+
+    @classmethod
+    def get_group_for_user(cls, course_key, user, user_partition):
+        checkpoint = user_partition.parameters["location"]
```

Comment:
@wedaly do we need to consider the case in with `VerificationPartitionScheme` don't have `location` in `parameters`?